Automattic/wp-calypso
# Add WooCommerce onboarding styles to the WC Helper connection flow

Closes https://github.com/woocommerce/woocommerce-admin/issues/2600. This PR implements the Muriel WooCommerce onboarding designs to the helper OAuth login and sign-up flows. This uses a special query parameter, so other connection flows will remain the same for now.

Related (where this flow was implemented for the Jetpack connection): #33073, #34380.

#### Screenshots

<img width="1680" alt="Screen Shot 2019-08-07 at 11 06 42 AM" src="https://user-images.githubusercontent.com/689165/62636158-03224d00-b907-11e9-9b47-d4e78c17350a.png">

<img width="1680" alt="Screen Shot 2019-08-07 at 11 06 54 AM" src="https://user-images.githubusercontent.com/689165/62636177-09182e00-b907-11e9-866f-906b26c6cd7e.png">

<img width="1680" alt="Screen Shot 2019-08-07 at 11 07 09 AM" src="https://user-images.githubusercontent.com/689165/62636074-e4bc5180-b906-11e9-8357-2f368c7076bc.png">

#### Testing Directions

* Run `master` of https://github.com/woocommerce/wc-admin.
* In your local WordPress install, add `define( 'WOOCOMMERCE_CALYPSO_LOCAL', true );` to your `wp-config.php`.
* Logout of WooCommerce.com and WordPress.com (otherwise you may immediately end up on the authorization screen).
* Go to WooCommerce > Extensions > WooCommerce.com Subscriptions and disconnect from WooCommerce.com.
* Go to the dashboard and click the connection task.
	* If you don’t see the connection task:
		* Make sure you have skipped the profiler and have `items_purchased` set to true: https://gist.github.com/justinshreve/da09cc57ec8e0d07ac8e919cef7a926c. 
		* In WooCommerce Admin, edit `client/dashboard/index.js` and set `requiredTasksComplete` to false.
* If you have the above patches applied, you should end up on the onboarding themed login page. Verify things look correct, you can login, and that you can also go to the signup form.
* Verify things look correct on the signup form, you can signup, etc.
* You can change `wccom-from=onboarding` in the header to `wccom-from=cart` and view the cart stepper header.
* Logging in or signing up should redirect you to the unthemed WooCommerce.com connection page.

Additional testing:

In `wp-calypso` edit the `config/development.json` feature flag file and disable the new code by setting `woocommerce/onboarding-oauth` to false. Restart Calypso. You can test the connection again and verify that you see the existing OAuth styles.

#### Additional Notes/Questions
* During a rebase, the Apple sign-in button was added. This area is starting to feel a bit like Nascar with multiple sign-in buttons that take up the same space as the normal login field. I left it alone for now, but was wondering if we want to do something in our flow @jameskoster? We can address this in a follow-up PR.
* The lost password flow still has WP.com colors. I will create a separate issue for this as it will require more work to figure out how to get this to work per the designs.
